src/ycc_hull/controllers/notifications/helpers_notifications_controller.py

```diff
@@ -1,28 +1,239 @@
+        self._title_change = self._diff.pop("title", None)
+        self._short_description_change = self._diff.pop("shortDescription", None)
+        self._long_description_changed = (
+            self._diff.pop("longDescription", None) is not None
+        )
+        self._contact_changed = False
+        self._timing_changed = False
+        self._captain_required_licence_change = self._diff.pop(
+            "captainRequiredLicenceInfo.licence", None
+        )
+        self._helper_min_max_count_changed = False
+        self._urgent_changed = self._diff.pop("urgent", None) is not None
+        self._published_changed = self._diff.pop("published", None) is not None
+        self._captain_changed = False
+        self._helpers_changed = False
+        self._status_changed = False
+
+        self.summary: list[str] = []
+        self.relevant_details: dict[str, Any] = {}
+
+        self._detect_changes()
+        self._compute_known()
+        self._collect_undetected_changes()
+
+    def _add_detailed_change(self, label: str, previous_value: Any) -> None:
+        self.summary.append(label)
+        self.relevant_details[f"Previous {label}"] = previous_value
+
+    def _detect_changes(self) -> None:
+        # Group fields covering the same concept and remove keys we never want to report
+        for remaining_key in list(self._diff.keys()):
+            if (
+                remaining_key == "id"
+                or remaining_key.endswith(".id")
+                or remaining_key.endswith("Id")
+            ):
+                self._diff.pop(remaining_key, None)
+            elif remaining_key.startswith("category."):
+                self._diff.pop(remaining_key, None)
+            elif remaining_key.startswith("contact."):
+                self._contact_changed = True
+                self._diff.pop(remaining_key, None)
+            elif (
+                remaining_key == "startsAt"
+                or remaining_key == "endsAt"
+                or remaining_key == "deadline"
+            ):
+                self._timing_changed = True
+                self._diff.pop(remaining_key, None)
+            elif remaining_key == "helperMinCount" or remaining_key == "helperMaxCount":
+                self._helper_min_max_count_changed = True
+                self._diff.pop(remaining_key, None)
+            elif remaining_key.startswith("captain."):
+                self._captain_changed = True
+                self._diff.pop(remaining_key, None)
+            elif remaining_key.startswith("helpers."):
+                self._helpers_changed = True
+                self._diff.pop(remaining_key, None)
+            elif (
+                remaining_key.startswith("marked_as_done_")
+                or remaining_key.startswith("validated_")
+                or remaining_key.startswith("validation_")
+            ):
+                self._status_changed = True
+                self._diff.pop(remaining_key, None)
```

Comment:
probably good to add an extra else: log (no condition for this key) for debugging?

Reply:
undetected keys are checked, collected and logged by another function